### ISSUE TITLE
Failed to parse configuration options: --module-syntax

### ISSUE THREAD

**Issue by omula** [easybuilders/easybuild-framework#2564]
Parameter `--module-syntax` seems to be ignored.

Log output: https://gist.github.com/omula/62ce29ea4943e1303f1f75fc7f160ebb

OS: Centos 6.4
```
$ python --version
Python 2.6.6
```

**Comment by boegel** [comment]
The error:

```
ERROR: Failed to parse configuration options: 'Generating Lua module files requires Lmod as modules tool; use --module-syntax to specify a different module syntax to use (Lua, Tcl)'
```

Relevant info from debug log:

```
== 2018-09-05 18:55:01,455 generaloption.py:1263 DEBUG parseoptions: options from environment ['--installpath=/home/software/RHEL/6/Nehalem', '--modules-tool=EnvironmentModulesC', '--prefix=/home/software/RHEL/6/Nehalem', '--sourcepath=/home/software/sources', '--tmp-logdir=/home/software/tmp']
== 2018-09-05 18:55:01,455 generaloption.py:1264 DEBUG parseoptions: options from commandline ['--module-syntax=Tcl', '--show-config']
== 2018-09-05 18:55:01,455 generaloption.py:1272 DEBUG Found options {'module_syntax': 'Tcl',  'modules_tool': 'EnvironmentModulesC', ...} args []
```

Relevant code from `easybuild/tools/options.py`:

```python
if self.options.module_syntax == ModuleGeneratorLua.SYNTAX and self.options.modules_tool != Lmod.__name__:
```

(see https://github.com/easybuilders/easybuild-framework/blob/master/easybuild/tools/options.py#L777)

The log clearly shows that the error doesn't make any sense...

`ModuleGeneratorLua.SYNTAX` is equivalent to `'Lua'`, which is not `'Tcl'`, so the error can never be triggered...

Is this a vanilla EasyBuild framework installation @omula?!

**Comment by boegel** [comment]
@omula With EasyBuild 3.6.2, you should have these MD5 checksums:

```
$ md5 easybuild/tools/options.py easybuild/tools/module_generator.py
MD5 (easybuild/tools/options.py) = 7b74862fc3269ef84d2de600212a1988
MD5 (easybuild/tools/module_generator.py) = 993838f773940d6535781e4ab6a89358
```

**Comment by boegel** [comment]
After a bit of back & forth with @omula on Slack, it turns out this is a bug in `--show-config`, or at least in the way in which the checks that are done in `postprocess` in `options.py` interact with `--show-config`...

The problem is that when `--show-config` is used, we reprocess the configuration on each of the different configuration levels separately (so we can tell what is configured on which level).
Therefore, if `--module-syntax Tcl` is used in combination with `export EASYBUILD_MODULES_TOOL=EnvironmentModulesC` (or with `modules_tool` set in a configuration file), it triggers and error when the configuration specified in the environment is parsed because `Lua` is the default for `--module-syntax`...

I'll look into a fix for this. Until that's in place, a workaround is to configure the modules tool & syntax on the same configuration level (e.g. both in the environment).

**Comment by boegel** [comment]
reproduced in tests + fixed in #2567
